tests/integration/s2n_tls13_handshake_tests.py

```diff
@@ -48,30 +47,41 @@ def verify_hrr_random_data(server, client):
             break
 
     return result
-    
-def key_update_recv(server, client):
+
+def key_update_test(server, client):
     '''
-    This test checks that a key update can be processed by s2n. It runs three times to prove that s2n can
-    process several key updates in a row.
+    This test proves that both server and client traffic keys can be successfully updated 
+    in a conversation between Openssl and s2n. It runs three times to confirm that multiple
+    keyupdates can be performed in a row. 
     '''
     result = Result()
     result.status = Status.PASSED
     for i in range(3):
-        msg = "Message:" + str(uuid.uuid4())
-        client.stdin.write(("k\n\n").encode("utf-8"))
+        openssl_msg = "Message:" + str(uuid.uuid4())
+        s2n_msg = "Message:" + str(uuid.uuid4())
+        # 'K' triggers Openssl to send a KeyUpdate message
+        client.stdin.write(("K\n\n").encode("utf-8"))
         client.stdin.flush()
-        line = ''
         # Confirm that the keyupdate was sent
+        line = ''
         while('KEYUPDATE' not in line):
```

Comment:
Good call-out. The reason why I added this while loop is that the 'KEYUPDATE' string takes about .1 seconds to appear. I could switch this out for something like: 
time.sleep(0.1);
if not (wait_for_output(client, 'KEYUPDATE', 100)):
            result.status = Status.FAILED;
I thought it would be better to avoid the sleep statement which is why I chose the while loop.

Reply:
My concern here is that if the test fails in the right way, could this turn into an infinite loop and cause the test to hang?